File: SuperGLU/Core/Config/Tests/ServiceLauncherTest.java
package Core.Config.Tests;

import org.junit.Assert;
import org.junit.Test;

import Core.Config.ServiceConfigurations;
import Core.Config.ServiceLauncher;

public class ServiceLauncherTest {

	@Test
	public void testLaunchAndConnectAllServices() {
		ServiceLauncher launcher = new ServiceLauncher();
		
		ServiceConfigurations configurations = launcher.readConfigurationFromFile("./SuperGLU/Core/Config/Tests/agentsConfig.json");
		
		launcher.launchAndConnectAllServices(configurations);
		
		
		Assert.assertEquals(2, launcher.getServices().size());
		
		launcher.stopService("socketIOGateway");
	}

	@Test
	public void testReadConfigurationFromFile() {
		ServiceLauncher launcher = new ServiceLauncher();
		
		ServiceConfigurations configurations = launcher.readConfigurationFromFile("./SuperGLU/Core/Config/Tests/agentsConfig.json");
		
		Assert.assertEquals(2, configurations.getServiceConfigurationMap().size());
		
		
	}
	
	
	@Test
	public void testShutdownActiveService()
	{
		ServiceLauncher launcher = new ServiceLauncher();
		
		ServiceConfigurations configurations = launcher.readConfigurationFromFile("./SuperGLU/Core/Config/Tests/agentsConfig.json");
		
		launcher.launchAndConnectAllServices(configurations);
		
		launcher.stopService("socketIOGateway");
	}

}

File: python_module/SuperGLU/Core/Tests/ServiceLauncherTest.py
'''
Created on Mar 13, 2018

@author: auerbach
'''
import unittest
from SuperGLU.Core.MessagingGateway import BaseMessagingNode
from SuperGLU.Core.ServiceConfiguration import ServiceConfiguration
from SuperGLU.Core.ServiceLauncher import ServiceLauncher
from SuperGLU.Util.Serialization import makeSerialized, serializeObject
from SuperGLU.Util.Serialization import JSON_STANDARD_FORMAT


class MockService1(BaseMessagingNode):
    
    def __init__(self, serviceConfiguration):
        super(MockService1, self).__init__(serviceConfiguration.getId(), None, None, None, serviceConfiguration.getBlackList(), serviceConfiguration.getWhiteList())
        

class ServiceLauncherTest(unittest.TestCase):


    def testReadFile(self):
        launcher = ServiceLauncher()
        config = launcher.readConfigurationFromFile("tests/agentsConfig.json", None)
        print (serializeObject(config, JSON_STANDARD_FORMAT))
        
    def testLaunchServices(self):
        launcher = ServiceLauncher()
        config = launcher.readConfigurationFromFile("tests/agentsConfig.json", None)
        launcher.launchAndConnectAllServices(config)
if __name__ == "__main__":
    #import sys;sys.argv = ['', 'Test.testName']
    unittest.main()